_rules/SC4-1-2-aria-state-or-property-has-valid-value.md

```diff
@@ -0,0 +1,270 @@
+---
+name: ARIA state or property has valid value
+test_type: atomic
+
+description: |
+   This rule checks that each ARIA state or property has a valid value
+   
+success_criterion:
+- 4.1.2 # Name, Role, Value
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Wilco Fiers
+- Anne Thyme Nørregaard
+---
+
+## Test procedure
+
+### Applicability
+
+Any [non-empty](#non-empty) [WAI-ARIA 1.1 state or property](https://www.w3.org/TR/wai-aria-1.1/#state_prop_def) that is specified on an HTML or SVG element.
+
+### Expectation
+
+Each test target has a valid value according to its [WAI-ARIA 1.1 value type](https://www.w3.org/TR/wai-aria-1.1/#propcharacteristic_value).
+
+For value types `ID Reference` and `ID Reference List`, a valid value for a [WAI-ARIA required property](https://www.w3.org/TR/wai-aria-1.1/#requiredState) requires that the element(s) with the given id(s) exists in the same [document tree](https://www.w3.org/TR/dom41/#document-trees) or [shadow tree](https://www.w3.org/TR/dom41/#shadow-trees) as the element that specifies the target attribute. For non-required properties, this is not a requirement.
```

Comment:
I disagree that this should be required. There are good use cases for using IDs of elements that don't yet exist. An example I came across pretty recently was one where aria-errormessage had a static value, and by adding or removing the element with that ID would they toggle the error message.

Reply:
This is only a requirement for required properties, as also mentioned in the last sentence. If an `aria-controls` property is required, for example, it makes little sense to consider an empty value invalid and then on the other hand consider a value that points to a non-existing ID valid as the two are essentially equivalent.